src/main/java/org/apache/commons/compress/archivers/zip/ZipSplitOutputStream.java

```diff
@@ -42,14 +48,20 @@ final class ZipSplitOutputStream extends OutputStream {
     private static final long ZIP_SEGMENT_MIN_SIZE = 64 * 1024L;
     private static final long ZIP_SEGMENT_MAX_SIZE = 4294967295L;
 
-    private OutputStream outputStream;
+    private FileChannel currentChannel;
+    private FileRandomAccessOutputStream outputStream;
     private Path zipFile;
     private final long splitSize;
+    private long totalPosition;
     private int currentSplitSegmentIndex;
     private long currentSplitSegmentBytesWritten;
     private boolean finished;
     private final byte[] singleByte = new byte[1];
 
+    private List<Long> diskToPosition = new ArrayList<>();
+
+    private TreeMap<Long, Path> positionToFiles = new TreeMap<>();
```

Comment:
Is there going to be an issue here on case-insensitive file systems? Say, Windows, macOS?

Reply:
Not here, the name created and name re-opened for `writeFullyAt` will be the same.

The issue may be in the reader but it's rather the opposite - it may fail on case-sensitive filesystem if the names are not consistent.  Perhaps we can check both `.z01` and `.Z01` extensions.